README.md

```diff
@@ -305,6 +306,30 @@ method takes one argument:
 
 2. `chain` must return a value of the same Chain
 
+### ChainRec
+
+A value that implements the ChainRec specification must also implement the Chain specification.
+
+1. `T.chainRec((done, next, value) => g(done(value)), i) `is equivalent to `g(i)`
```

Comment:
Still something bugs me about type of `g`. It seems like for any law in the spec where we have a function like that we can say "for any `g` of certain type". For example:

> `u.map(x => f(g(x)))` is equivalent to `u.map(g).map(f)`

for any:
- `u :: f a`
- `g :: a -> b` 
- `f :: b -> c`

> `a.of(f).ap(a.of(x))` is equivalent to `a.of(f(x))`

for any:
- `x :: a`
- `f :: a -> b`

> `m.chain(f).chain(g)` is equivalent to `m.chain(x => f(x).chain(g))`

for any:
- `m :: m a`
- `f :: a -> m b`
- `g :: b -> m c`

But here we can't say something like that about `g` because we can't identify it's type. I can't quite articulate why this is a problem, but it feels wrong.